text/0000-support-single-implementation.md

```diff
@@ -0,0 +1,131 @@
+
+### Migrating from the current architecture
+
+To the extent that the current architecture has already been implemented, a migration path may look something like:
+
+1. Start with the "minimal implementation"
+1. Send data to the particular event's exporter
+1. Make the "SDK" conform to the new exporter implementation
+1. Break out "transformer" logic from the SDK
+
+## Trade-offs and mitigations
+
+### Providing enough flexibility for vendors
+
+The biggest risk with this RFC is probably the question of whether it supports the needed flexibility.
+
+For potential future third-party needs that cannot be addressed with an exporter (or even a hook-supplemented exporter), the solution is likely to **add the functionality into the API** - particularly where multiple parties have the same or a similar need.
+
+See, for example, the [request for "transaction" support in OpenCensus](https://github.com/census-instrumentation/opencensus-specs/issues/229). A feature like this would be a strong candidate for getting added directly into the OpenTelemetry core.
+
+### Migration costs
+
+Given that OpenTelemetry is still very young and not yet production-ready, migration costs should be relatively small for both end users and implementors - particularly to the extent that the ambiguity between the API and SDK has slowed down implementation.
+
+## Prior art and alternatives
+
+### Prior art
+
+The three main prior art examples are OpenTracing, OpenCensus, and the existing OpenTelemetry architecture.
+
+**OpenTracing** handles export functionality by having each vendor (or other party) reimplement the whole API. While this maximizes the flexibility for vendors, it also means that everyone must redo a lot of the same logic - including any logic that may be considered vendor-agnostic.
+
+**OpenCensus** handles it by having a single implementation which then sends data to tracing and stats exporters. _(**Note:** While architecturally this is very similar to the RFC's proposal, key differences include OpenCensus itself being quite opinionated and sometimes stateful, depending on the implementation.)_
+
+The **existing OpenTelemetry architecture** essentially combines the OpenTracing and OpenCensus approaches. As with OpenTracing, the core API provides an interface that can then have multiple implementations. The equivalent to the OpenCensus "opinionated implementation" is then provided as a *particular* implementation (the SDK), which in turn supports exporters.
+
+### Alternatives
+
+What are some prior and/or alternative approaches? For instance, is there a corresponding feature in OpenTracing or OpenCensus? What are some ideas that you have rejected?
+
+## Open questions
+
+What are some questions that you know aren't resolved yet by the RFC? These may be questions that could be answered through further discussion, implementation experiments, or anything else that the future may bring.
```

Comment:
An alternative – and, frankly, one I would prefer – would be to maintain the separation between API and implementation but commit to a "standard" or "recommended" implementation that those of us on the ground floor of OpenTelemetry would all (most likely) use, at least "for now."

As small as a reified Span object sounds, in practice it will be the place that certain optimizations and quirks live, so I there is a meaningful "unknown unknown" here about forwards-compatibility, both with optimizations and features we haven't thought of yet, and also with vendors who are not participating in these early days but will participate later on in 2020+.